4.component.md

```diff
@@ -491,156 +70,49 @@ metadata:
-    - name: USERNAME
-      fromParam: 'username'
-    - name: PASSWORD
-      fromParam: 'password'
-    - name: BACKEND_ADDRESS
-      fromParam: 'backend-address'
-    livenessProbe:
-      httpGet:
-        port: 8080
-        path: /healthz
-    readinessProbe:
-      httpGet:
-        port: 8080
-        path: /healthz
+    fieldPaths:
+    - "spec.containers[0].env[0].value"
 ```
 
-Our second component implements both the backend for our Twitter bot's
-administrative interface and a background worker process that will Tweet
-end-user-defined content at end-user-defined intervals (defined through the
-administrative interface). This component also persists its end-user-defined
-configuration to disk as a JSON file.
-
-This component's design prevents it from scaling horizontally so we should only
-have at most _one_ instance of it running. It also has an endpoint for an administrative interface. As such, this component is best described
-as a `SingletonServer`. Connection details for the Twitter API are
-configurable. The component schematic also describes the file system location
-where the component persists end-user-defined configuration.
-
-```yaml
-apiVersion: core.oam.dev/v1alpha1
-kind: ComponentSchematic
-metadata:
-  name: admin-backend
-  annotations:
-    version: v1.0.0
-    description: >
-      Sample component schematic that describes the backend for our Twitter bot.
-spec:
-  workloadType: core.oam.dev/v1.SingletonServer
-  osType: linux
-  parameters:
-  - name: twitter-consumer-key
-    description: Twitter API consumer key
-    type: string
-    required: true
-  - name: twitter-consumer-secret
-    description: Twitter API consumer secret
-    type: string
-    required: true
-  - name: twitter-access-token
-    description: Twitter API access token
-    type: string
-    required: true
-  - name: twitter-access-token-secret
-    description: Twitter API access token secret
-    type: string
-    required: true
-  containers:
-  - name: my-twitter-bot-backend
-    image:
-      name: example/my-twitter-bot-backend:1.0.0
-      digest: sha256:6c3c624b58dbbcd3c0dd82b4c53f04194d1247c6eebdaab7c610cf7d66709b3b
-    resources:
-      cpu:
-        required: 1.0
-      memory:
-        required: 100MB
-      volumes:
-      - name: config
-        mountPath: /var/lib/my-twitter-bot/conf
-        accessMode: RW
-        sharingPolicy: Exclusive
-    ports:
-    - name: http
-      value: 8080
-    env:
-    - name: TWITTER_CONSUMER_KEY
-      fromParam: 'twitter-consumer-key'
-    - name: TWITTER_CONSUMER_SECRET
-      fromParam: 'twitter-consumer-secret'
-    - name: TWITTER_ACCESS_TOKEN
-      fromParam: 'twitter-access-token'
-    - name: TWITTER_ACCESS_TOKEN_SECRET
-      fromParam: 'twitter-access-token-secret'
-    livenessProbe:
-      httpGet:
-        port: 8080
-        path: /healthz
-    readinessProbe:
-      httpGet:
-        port: 8080
-        path: /healthz
-```
-
-#### Example: Extended Workload Type
```

Comment:
we also need to add an extended workload and component here

Reply:
We don't have a good extended workload definition (schema) and I don't want to add a CRD example in our spec